### ISSUE TITLE
Review EdgeX Architecture

### ISSUE THREAD

**Issue by mmccool** [w3c/wot-usecases#45]
It would be useful to review architectures intended to support similar use cases to WoT (or that are targets for WoT integration) in order to confirm that we are meeting their requirements.
This issue proposes that we do this for EdgeX, which is basically a "WoT Gateway" (to use ITU-T terminology).   Here is a link to the docs for the "Fuji" release: https://fuji-docs.edgexfoundry.org/index.html

Some salient points:
- Security: Reverse proxy (Kong) "API Gateway", JWT, OAuth2, https/TLS, secret store (Vault), etc; however I have created a separate issue in the security repo (https://github.com/w3c/wot-security/issues/180) to review the EdgeX security architecture
- Metadata repository service, which I think is the main point where we can collaborate.  Right now the metadata seems fairly limited, it seems for example that properties are just single numbers not structured data. 
- There is also a Core Data repository service that provides a historical database of values (a service we do not consider in our current architecture)

Probably the deliverable here should be someone from WoT doing an in-depth reading of the spec and maybe putting together a summary presentation, followed by inviting someone from EdgeX (eg Brad Corrian) to have a discussion and get input.   In parallel I am also putting together a proposal to EdgeX on WoT integration so they can better support structured metadata.

**Comment by mmccool** [comment]
Sorry, I meant to put this issue under Architecture.   However, it does drive requirements if we consider "EdgeX integration" as a  very specific use case ;)

**Comment by mmccool** [comment]
Metadata does seem to be oriented around "device instance" (similar to Thing Descriptions) not "classes" (eg Thing Models).  Metadata service also seems to line up in functionality with what we are working on in the Discovery TF, so maybe I should also make a more specific issue to look at the EdgeX discovery architecture.

**Comment by mlagally** [comment]
@mmccool please carry over this issue into architecture and close here.

**Comment by mlagally** [comment]
UC call on May 4th:
Need to carry over and add a hub section in the architecture document.

**Comment by mlagally** [comment]
See architecture issue: https://github.com/w3c/wot-architecture/issues/585

**Comment by mlagally** [comment]
UC call on May 11th:
Hub architecture is very similar to WebThings, some details probably are different.
Need to reengage EdgeX colleagues and talk to Krellian, ideally doing a prototype for an upcoming plugfest or POC.
